src/components/ListItem.jsx

```diff
@@ -1,5 +1,57 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+const currentTime = new Date();
+const numOfSecondsinMin = 60;
+const millisecondsToSeconds = 1000;
+
+export function ListItem({ name, item, listToken }) {
+	const { isChecked, id, totalPurchases, dateLastPurchased } = item;
+
+	if (isChecked) {
+		//check if it's been 24 hours since item was last marked as purchased
+		const timeDiffMinutes =
+			(currentTime.getTime() / millisecondsToSeconds -
+				dateLastPurchased.seconds) /
+			numOfSecondsinMin;
+		const minutesBeforeReset = 1440; // 24 hours x 60 minutes per hour
+		if (timeDiffMinutes >= minutesBeforeReset) {
+			const itemData = {
+				isChecked: false,
+			};
+			updateItem(listToken, id, itemData);
+		}
+	}
+
+	const handlePurchase = () => {
+		if (isChecked) {
+			const itemData = {
+				isChecked: false,
```

Comment:
This is where the user can uncheck the box if it was a mistake, right? So the isChecked value reverts to false. But what about the other itemData? It seems that persists in the database. Which probably doesn't matter too much yet, but when we get to working with the future purchase date and anything involving the number of items purchased, it will throw things off.

Reply:
Hey @tmflower, Weston and I discussed this as well. Being able to give the user the option to uncheck an item was totally my idea (not an acceptance criterion) and Weston was kind enough to oblige. haha. We did notice that this may cause future issues with the date last purchased and those calculations but decided not to tackle this issue at this time since the other ticket items were quite heavy. I can add this as a to-do item so that it can be addressed in the upcoming weeks.

Reply:
That's a great point, @tmflower. It would be simple enough to decrement the count of the total purchases, however, we would no longer have access to the previous `dateLastPurchased`. I was looking ahead at other issues to determine if setting `dateLastPurchased` back to `null` would have any consequences, and it looks like #10 is going to calculate the `dateNextPurchased` when we check the box. I worry that this unchecking is going to have unintended consequences. The only workaround I can think of is storing `dateLastPurchased` and `dateNextPurchased` as arrays, so we maintain a history of past values. This way if a user unchecks the 'purchased' checkbox, we just have to decrement the total purchase count and pop/remove the last array entry on last/next purchased... Thoughts?

Reply:
@wheninseattle So, one possible way to avoid having to deal with the complications of unintended consequences could be to add a "confirm" or "are you sure?" (or the threatening, "this action cannot be reversed!" ⛔️ 😆) button for the user to click upon checking the box, BEFORE the function runs. Not a failsafe but an option.

Other than that, something along the lines of what you described was what came to mind for me: total purchase count is an easy adjustment, but `dateLastPurchased` less so. It shouldn't necessarily be set back to null because the accidental checkmark could happen at any time (i.e. not just as the first purchase). So it makes sense as you said to somehow store that date in short term memory so it could be retrieved if needed. And as you mentioned, eventually the `dateNextPurchased` would be affected too. And then what if there are multiple checks and unchecks by that wayward user 😼 that just likes to click things at random and try to break them? Does each item that is clicked need to be held in short term memory as an object with its associated data? Yikes! Maybe this one goes on that optional features list?

Reply:
Hmm, that is a good thought, although, I worry that asking for confirmation on every purchase could be frustrating to the user. I'm inclined to agree that the uncheck could go on our bonus feature list, and I'd want Ar'Niqua to weigh in. What do you think @a-cole55?

Reply:
Now this is where we can get ourselves in a bit of a pickle. Technically you have hit that acceptance criteria of once it's checked, it updates the Firebase db. What you have worked is functional and does hit all the AC. 

However, just like @tmflower mentioned if it's by accident then the Firebase now has updated with the new 'incorrect' timestamp. Also, just like @wheninseattle mentioned, there will be consequences because the data in the db is no longer accurate. I know in the acceptance criteria, it doesn't mention what to do with the uncheck function, but this ties into what you are working on. 

Unfortunately the current state of your code is now a bug since the db will now carry inaccurate information. Pushing the current code to main branch may cause the next team/developers roadblock especially for the next team working on #10. 

So currently I can't accept the code until the Firebase db can be updated accurately.

Reply:
Great comment @YolandaHaynes. @a-cole55 and I decided to avoid the pickle all together. We have removed the option to uncheck the box, and created an optional TODO to handle the uncheck in the future. Thanks!